includes/wpml/class-course-progress.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * File containing \Sensei\WPML\Course_Progress class.
+ *
+ * @package sensei
+ */
+
+namespace Sensei\WPML;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Course_Progress
+ *
+ * Compatibility code with WPML.
+ *
+ * @since $$next-version$$
+ *
+ * @internal
+ */
+class Course_Progress {
+	/**
+	 * Init hooks.
+	 */
+	public function init() {
+		add_filter( 'sensei_course_is_user_enrolled_course_id', array( $this, 'translate_course_id' ), 10, 1 );
+		add_filter( 'sensei_block_take_course_course_id', array( $this, 'translate_course_id' ), 10, 1 );
+		add_filter( 'sensei_course_progress_create_course_id', array( $this, 'translate_course_id' ), 10, 1 );
+		add_filter( 'sensei_course_progress_get_course_id', array( $this, 'translate_course_id' ), 10, 1 );
+		add_filter( 'sensei_course_progress_has_course_id', array( $this, 'translate_course_id' ), 10, 1 );
+		add_filter( 'sensei_course_progress_delete_for_course_course_id', array( $this, 'translate_course_id' ), 10, 1 );
+		add_filter( 'sensei_course_progress_find_course_id', array( $this, 'translate_course_id' ), 10, 1 );
+		add_filter( 'sensei_lesson_progress_count_course_id', array( $this, 'translate_course_id' ), 10, 1 );
+		add_filter( 'sensei_course_start_course_id', array( $this, 'translate_course_id' ), 10, 1 );
+		add_filter( 'sensei_course_manual_enrolment_enroll_learner_course_id', array( $this, 'translate_course_id' ), 10, 1 );
+		add_filter( 'sensei_course_manual_enrolment_withdraw_learner_course_id', array( $this, 'translate_course_id' ), 10, 1 );
+	}
+
+	/**
+	 * Translate course ID.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @internal
+	 *
+	 * @param int $course_id Course ID.
+	 * @return int
+	 */
+	public function translate_course_id( $course_id ): int {
+		$course_id = (int) $course_id;
```

Comment:
We are not doing this explicit casting operation for lessons or quiz progress, is there any reason behind doing it for Course and Quiz?

Reply:
Ah, I think I encountered an error with course id at some point. Or, which is more likely Psalm argued about it.
Added explicit casting in all these cases for consistency: https://github.com/Automattic/sensei/pull/7492/commits/b9c61df1e32e3fd99245deb7e258ceb80debffd7
However, the reversal approach (not cast at all) is fine here as well.